internal/controller/operatorconfigmap_controller.go

```diff
@@ -1075,3 +1089,74 @@ func parseTolerations(csiPluginTolerations string) ([]corev1.Toleration, error)
 	}
 	return tolerations, nil
 }
+
+func (c *OperatorConfigMapReconciler) removeNoobaa() error {
+	noobaa := &nbv1.NooBaa{}
+	noobaa.Name = "remote-noobaa"
+	noobaa.Namespace = c.OperatorNamespace
+
+	if err := c.get(noobaa); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to get remote noobaa: %v", err)
+	} else if noobaa.UID != "" && noobaa.GetDeletionTimestamp().IsZero() {
+		index := slices.IndexFunc(
+			noobaa.GetOwnerReferences(),
+			func(ref metav1.OwnerReference) bool {
+				return ref.Kind == "StorageClient"
+			},
+		)
+		if index == -1 {
+			if err := c.delete(noobaa); err != nil {
+				return fmt.Errorf("failed to delete remote noobaa: %v", err)
+			}
+		}
+	}
+	return nil
+}
+
+func (c *OperatorConfigMapReconciler) removeNoobaaOperator() error {
+
+	nb := &metav1.PartialObjectMetadataList{}
+	nb.SetGroupVersionKind(nbv1.SchemeGroupVersion.WithKind("NooBaa"))
+	if err := c.list(nb); err != nil {
```

Comment:
Please cap the list with count = 1. if not you will load all noobaa crs from the k8s server